src/pages/search.tsx

```diff
@@ -33,28 +33,39 @@ const Search: React.FC = () => {
     const [currentPage, setCurrentPage] = useState(1);
     const [totalPages, setTotalPages] = useState(1);
 
-    const searchTeacher = async (value: string) => {
-        if (value.trim() && searchedValue.current !== value) {
-            try {
-                const data = await searchByInput(value);
-                setTeachers(data.data);
-                searchedValue.current = value;
-
-                setCurrentPage(data.paging.pageNumber);
-                setTotalPages(data.paging.pageCount);
-            } catch (e) {
-                console.error(e);
-            }
-        }
-    };
-
+    /**
+     * @description Initial loading of data, reflecting the current search state in URL.
+     * Will determine type of search from query params set searched value to the input field.
+     */
     useEffect(() => {
         const searchString = createSearchString(searchStateMode || '', searchStateInput || '');
-        onSubmit(searchString);
-    }, [searchStateInput, searchStateMode]);
 
+        setSearchValue(searchString);
+
+        if (inputRef.current) {
+            inputRef.current.select();
+        }
+    }, [searchStateInput, searchStateMode, router.isReady]);
+
+    /**
+     * @description Search for teachers on every change of search value.
+     * Search value expected to be updated on search form submission.
+     */
     useEffect(() => {
-        searchTeacher(searchValue);
+        (async () => {
+            if (searchValue.trim() && searchedValue.current !== searchValue) {
+                try {
+                    const data = await searchByInput(searchValue);
+                    setTeachers(data.data);
+                    searchedValue.current = searchValue;
+
+                    setCurrentPage(data.paging.pageNumber);
+                    setTotalPages(data.paging.pageCount);
+                } catch (e) {
+                    console.error(e);
+                }
+            }
+        })();
```

Comment:
removed searchTeacher function, to clear up component code. That function called only in that hook, so I propose to keep it all in one place, to prevent misuse of state variables